### DISCUSSION TITLE
changing the staticDir on the fly

### DISCUSSION THREAD

**Discussion by ylahav** [payloadcms/payload#3416] [Q&A]
I am writing a plugin that will support categories and tags on media files.
Beacuase I ave a lot of files (more than 1k) I want to sparate between them using sub-directoris based on the category value...
Can I change the staticDir value just before uploading the image? or I have to define my own "upload" field?

**Comment by jentzheng** [comment]
I would like to attract some idea, seems it's necessary to support subfolder in multi-tenant platform

```
export const Media: CollectionConfig = {
  slug: 'media',
  upload: {
    staticDir: (doc)=> `media/${doc.tenant.name}`,
  },
    fields: [
    {
      name: 'caption',
      type: 'text',
    },
    tenant, <---------- multi tenant collection
  ],
}

```

**Reply by SarukaUsagi** [reply]
Same here, seems really important for the multi-tenant features
